### PR TITLE
UNG-2551 - Adding to readiness check status of creator loops

### PR DESCRIPTION
**Pull request by wojciechUrbanski**

### CODE REVIEW

File: src/main/scala/com/ubirch/services/healthcheck/HealthCheckService.scala [hunk 1]
```diff
@@ -68,6 +80,23 @@ class DefaultHealthCheckService @Inject() (
     }
   }
 
+  private def performLoopCreationHealthCheck(state: AtomicAny[LoopState], loopName: String) = Task(state.get() match {
+    case common.Running =>
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/healthcheck/HealthCheckService.scala:84]: When OOM occurs then the loop stops, the status is still Running?

File: src/main/scala/com/ubirch/services/poc/PocAdminCreationLoop.scala [hunk 2]
```diff
@@ -20,17 +25,27 @@ class PocAdminCreationLoopImpl @Inject() (conf: Config, pocAdminCreator: PocAdmi
...
+        logger.error("some unexpected error occurred during poc admin creation", ex)
+        retryWithDelay(source).delayExecution(pocAdminCreatorInterval.seconds)
     }
   }
 
   override def startPocAdminCreationLoop[T](operation: PocAdminCreationResult => Observable[T]): Observable[T] =
-    retryWithDelay(startPocAdminCreation.flatMap(operation))
+    retryWithDelay(startPocAdminCreation.flatMap(operation)).guaranteeCase {
+      case ExitCase.Canceled  => Task(PocAdminCreationLoop.loopState.set(Cancelled))
+      case ExitCase.Error(_)  => Task(PocAdminCreationLoop.loopState.set(ErrorTerminated))
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/poc/PocAdminCreationLoop.scala:44]: Ah, so when the Fatal error occurs, can we catch this error here?

File: src/main/scala/com/ubirch/services/healthcheck/HealthCheckService.scala [hunk 3]
```diff
@@ -68,6 +91,51 @@ class DefaultHealthCheckService @Inject() (
...
+      UnhealthyService
+    case common.ErrorTerminated(dateTime) =>
+      logger.error(s"$loopName is terminated because unexpected error has happened at ${dateTime.toDateTimeISO}")
+      UnhealthyService
+    case common.Completed =>
+      logger.error(s"$loopName has been completed")
+      UnhealthyService
+  })
+
+  private def howLongFromNow(from: DateTime) = {
```
Comment (Hiroki6) [inline src/main/scala/com/ubirch/services/healthcheck/HealthCheckService.scala:135]: Please add return type.

File: src/main/scala/com/ubirch/services/healthcheck/HealthCheckService.scala [hunk 4]
```diff
@@ -20,9 +28,16 @@ class DefaultHealthCheckService @Inject() (
   keycloakCertifyConfig: KeycloakCertifyConfig,
   keycloakDeviceConfig: KeycloakDeviceConfig,
   keycloakConnector: KeycloakConnector,
-  teamDriveClient: TeamDriveClient
+  teamDriveClient: TeamDriveClient,
+  conf: Config
 ) extends HealthCheckService
   with LazyLogging {
+
+  private val elementsProcessingTimeout = conf.getInt(ConfPaths.HealthChecks.Timeouts.ELEMENTS_PROCESSING).minutes
```
[the remark this answers is not in the document]
Comment (wojciechUrbanski) [reply src/main/scala/com/ubirch/services/healthcheck/HealthCheckService.scala]: Done :)

File: src/main/scala/com/ubirch/services/healthcheck/HealthCheckService.scala (hunk 3 above)
Comment (wojciechUrbanski) [reply src/main/scala/com/ubirch/services/healthcheck/HealthCheckService.scala]: Done :)

**Review by Hiroki6 — APPROVED** [review]
thank you!

**Pull request merged**